### PR TITLE
ValueSets and spec updates

### PR DESCRIPTION
**Pull request by mgramigna**
~Depends on https://github.com/projecttacoma/fqm-execution/pull/186 since it uses the newly-exported ValueSetResolver class with the updated VSAC URL~

~For this to be merged, a new patch version with the linked changes will be released and brought into this code as a dependency.~

# Summary

This PR includes changes worked on at the January '23 connectathon. Mainly, support for using the VSAC FHIR API during measure packaging, which is useful for the end-to-end workflow demonstration of knowledge repository queries all the way through data submission and measure evaluation.

Besides that, some various QOL updates are included here, such as typescript updates for a custom capability statement and fixes to the period start/end of data requirements to make them not required.

## New behavior

* When using the `include-terminology` query parameter, the server will resolve any missing ValueSets and cache them in the file system.
  * Looking for feedback on the disabling of caching in unit tests and if that should change in any way
* periodStart/end are omitted from the calculation options of fqm-execution if they are not provided in the query params for data requirements
  * Technically this will trigger the default 2019 behavior of fqm-execution, but that will be addressed in separate tasking.

## Code changes

* Bring in valueset resolver and use it in packaging when include-terminology is true.
* Conditionally add periodStart/end based on if they're provided
* Add unit tests for valueset resolution
* Update node-fhir-server-core type defs to adhere to [custom capability docs](https://github.com/bluehalo/node-fhir-server-core/blob/master/docs/CustomCapability.md)

# Testing guidance

* To test valueset resolution, add your VSAC API key to .env (see .env.example)
* Run a $package operation with `include-terminology` set to `true` in the query parameters. It will take a long time
  * You'll see in the logs that it resolves and caches the valuesets
* Run the same request again. It should take basically no time at all since it uses a cache. Check the logs to confirm behavior

### CODE REVIEW

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Two small comments! Other than that, looks good!

Regarding potential caching unit tests, I see pros and cons of adding it. Definitely can't hurt to add it but may add more work than is necessary. @sarahmcdougall what do you think?

File: src/util/bundleUtils.ts [hunk 1]
```diff
@@ -86,6 +114,88 @@ export function getQueryFromReference(reference: string): Filter<any> {
   }
 }
 
+function hasNoDependencies(lib: fhir4.Library) {
+  return !lib.relatedArtifact || (Array.isArray(lib.relatedArtifact) && lib.relatedArtifact.length === 0);
+}
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply src/util/bundleUtils.ts:119]: I took it from deqm-test-server, and I believe it was a function there bc it was used in two places.

Looks like this exact statement appears when getting all the dependent libraries though, so i think we should use this function there two which justifies its existence ahha

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]
Yay works great and the caching is especially helpful! I also find the logger statements super helpful and you made a good call on what should be logged as info v. debug.

Only one comment but otherwise I think it's good to merge. In terms of disabling the caching in unit tests, I think that's fine. I saw that we have some unit tests in fqm-execution related to [ELM info caching ](https://github.com/projecttacoma/fqm-execution/blob/master/test/unit/helpers/ELMInfoCache.test.ts) so if it's easy to implement something similar to that, then go for it but otherwise I think the current unit tests are sufficient

File: test/util/bundleUtils.test.ts [hunk 2]
```diff
@@ -65,7 +129,13 @@ const MEASURE_WITH_LIBRARY: fhir4.Measure = {
 
 describe('bundleUtils', () => {
   beforeAll(() => {
-    return setupTestDatabase([LIB_WITH_DEPS, LIB_WITH_NO_DEPS, LIB_WITH_MISSING_DEPS]);
+    return setupTestDatabase([
+      LIB_WITH_DEPS,
+      LIB_WITH_NO_DEPS,
+      LIB_WITH_MISSING_DEPS,
+      LIB_WITH_EXTRA_VALUESET,
+      LIB_WITH_EXTRA_VALUESET
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply test/util/bundleUtils.test.ts:137]: This was a test and you passed

**Review by sarahmcdougall — APPROVED** [review]

**Review by elsaperelli — APPROVED** [review]
Lgtm! Holding off on merging until it's confirmed that you don't want to add caching unit tests.

**Review by elsaperelli — APPROVED** [review]
Lgtm again!

**Review by sarahmcdougall — APPROVED** [review]

**Pull request merged**